### DISCUSSION TITLE
Get NotificationTapped in Android on MAUI

### DISCUSSION THREAD

**Discussion by RpDoc-Christian** [thudugala/Plugin.LocalNotification#514] [Q&A]
Hello,

I tried to use your library for replacing our existing self written local notification code in our .NET 8 MAUI app. 

On iOS it worked fine on Android don't get the notification tapped event back in the app. The SplashActivity witch is mentioned in the wiki doesn't get it.

```
[Activity(Theme = "@style/Maui.SplashTheme",
    NoHistory = true,
    MainLauncher = true)]
public class SplashActivity : Activity
{
    protected override void OnCreate(Bundle savedInstanceState)
    {
        base.OnCreate(savedInstanceState);

        var actionId = Intent.GetIntExtra(LocalNotificationCenter.ReturnRequestActionId, -1000);
        var mainIntent = new Intent(Application.Context, typeof(MainActivity));

        mainIntent.SetFlags(ActivityFlags.SingleTop);
        if (actionId != -1000)
        {
            var data = Intent.GetStringExtra(LocalNotificationCenter.ReturnRequest);
            mainIntent.PutExtra(LocalNotificationCenter.ReturnRequestActionId, actionId);
            if (!string.IsNullOrEmpty(data))
            {
                mainIntent.PutExtra(LocalNotificationCenter.ReturnRequest, data);
            }
        }

        StartActivity(mainIntent);
    }

    protected override void OnResume()
    {
        base.OnResume();


    }

```

Can you give me some advice?

Sincerly

Christian

**Comment by thudugala** [comment]
@RpDoc-Christian can you please attach a sample project?

**Reply by RpDoc-Christian** [reply]
Hello,

I got it now here in MAUI 8 on Android, I get now the return message of the local notification even if the app is in background or was killed before. I do it in the main activity and I have no need for an extra SplashActivity on Android.

Your event NotificationActionTapped witch can be used in the cross platform part is nice, but it doesn't react when the app was killed and now starting back after touching a local notification. It would be nice more helpful that such an event handle all notification tapped situations on the several OS.

A project sample of my suggestion will be attached.

[LocalPushNotificationTestWithAndroidRecieving.zip](https://github.com/user-attachments/files/16599873/LocalPushNotificationTestWithAndroidRecieving.zip)

@thudugala Maybe you can do an update of your wiki documentation and your sample project about this issue.
